packages/types/src/utilities.ts

```diff
@@ -18,3 +18,9 @@ export type DeepPartial<T> = {
 }
 
 export type ValueOf<T> = T[keyof T];
+
+/**
+ * Based on https://stackoverflow.com/a/49725198
+ */
+export type RequireOnlyOne<T, Keys extends keyof T = keyof T> =
```

Comment:
What's this supposed to be used for? I'm not sure what `RequireOnlyOne` would mean in this context

Reply:
```ts
type Test RequireOnlyOne<{
   Obj1? any;
   Obj2? any;
   Obj3? any;
   Obj4? any;
}>
```

Only one of the Obj* can be set on a variable of test Test. Note that this already exist in mattermost-webapp @ `/utils/conditional_types.ts` but was not accessible from the packages.

Reply:
Yes indeed:

```ts
// clone of packages/mattermost-redux/src/types/actions.ts used for this test
export type ActionResult<Data = any, Error = any> = {
    data?: Data;
    error?: Error;
};

type ActionResultBis = RequireOnlyOne<ActionResult>;

const testOriginal: ActionResult = {
    data: 'test',
    error: 'test',
}; // <-- no error

const testWithRequireOnlyOne: ActionResultBis = {
    data: 'test',
    error: 'test',
}; // <-- error
```

Reply:
> In that case, would you mind just getting rid of utils/conditional_types and moving the other type here as well?

done!